antarest/storage/business/study_download_utils.py

```diff
@@ -0,0 +1,319 @@
+
+                if year not in matrix.data[area_name]:
+                    matrix.data[area_name][year] = dict()
+
+                matrix.data[area_name][year][column_name] = [
+                    row[index] for row in rows
+                ]
+
+        except (ChildNotFoundError, FilterError):
+            matrix.warnings.append(f"{area_name} has no child")
+
+    @staticmethod
+    def level_output_filter(
+        matrix: MatrixAggregationResult,
+        prefix: str,
+        year: int,
+        area_name: str,
+        study: Study,
+        url: str,
+        data: StudyDownloadDTO,
+    ) -> None:
+
+        if StudyDownloadLevelDTO.from_dict(data.level):
+            files_matcher = (
+                [f"values-{str(data.level)}", f"details-{str(data.level)}"]
+                if data.includeClusters
+                else [f"values-{str(data.level)}"]
+            )
+            for elm in files_matcher:
+                tmp_url = f"{url}/{elm}"
+                StudyDownloader.read_columns(
+                    matrix, prefix, year, area_name, study, tmp_url, data
+                )
+
+    @staticmethod
+    def apply_type_filters(
+        matrix: MatrixAggregationResult,
+        prefix: str,
+        year: int,
+        type_elm: Any,
+        elm: str,
+        study: Study,
+        url: str,
+        data: StudyDownloadDTO,
+        first_element_type_condition: bool,
+        second_element_type_condition: Callable[[str], bool],
+    ) -> None:
+
+        if first_element_type_condition:
+            if StudyDownloadType.from_dict(
+                data.type
+            ) == StudyDownloadType.LINK and isinstance(type_elm[elm], Area):
+                if type_elm[elm].links:
+                    for out_link in type_elm[elm].links.keys():
+                        if second_element_type_condition(out_link):
+                            tmp_prefix = f"{prefix}|{elm}^{out_link}"
+                            link_url = f"{url}/{out_link}"
+                            StudyDownloader.level_output_filter(
+                                matrix,
+                                tmp_prefix,
+                                year,
+                                f"{elm}^{out_link}",
+                                study,
+                                link_url,
+                                data,
+                            )
+            else:
+                tmp_prefix = f"{prefix}|{elm}"
+                StudyDownloader.level_output_filter(
+                    matrix, tmp_prefix, year, elm, study, url, data
+                )
+
+    @staticmethod
+    def get_filters(filter: str) -> Tuple[str, str]:
+        if filter and len(filter) > 0:
+            tmp_filters = filter.split(">")
+            if len(tmp_filters) >= 2:
+                return tmp_filters[0], tmp_filters[1]
+            else:
+                return filter, ""
+        return "", ""
+
+    @staticmethod
+    def select_filter(
```

Comment:
you can add a "data_type" argument (StudyDownloadType), so that it's clearer and easier to branch different filtering behavior